src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    this.forEach((item, index) => {
```

Comment:
Please check the task description
_Using Array methods is prohibited in this task. Use loops, index access, and array length if needed._